k9mail/src/main/java/com/fsck/k9/activity/ChooseSnooze.java

```diff
@@ -0,0 +1,228 @@
+        mTimeAndDayFormatter = new SimpleDateFormat(timeFormat, Locale.getDefault());
+        mTimeFormatter = android.text.format.DateFormat.getTimeFormat(this);
+        mRes = getResources();
+
+        getListView().setFastScrollEnabled(true);
+        getListView().setItemsCanFocus(false);
+        getListView().setChoiceMode(ListView.CHOICE_MODE_NONE);
+        Intent intent = getIntent();
+        String accountUuid = intent.getStringExtra(ChooseFolder.EXTRA_ACCOUNT);
+        mAccount = Preferences.getPreferences(this).getAccount(accountUuid);
+        mMessageReference = intent.getParcelableExtra(ChooseFolder.EXTRA_MESSAGE);
+
+        mAdapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1);
+        mAdapter.addAll(getTimes());
+
+        setListAdapter(mAdapter);
+
+        getListView().setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                SnoozeTime time = mAdapter.getItem(position);
+
+                if (time.timestamp <= 0) {
+                    // choose custom time from picker
+                    showDateTimePicker(new OnDateTimePickedListener() {
+                        @Override
+                        public void onDateTimePicked(long timestamp) {
+                            finishWithResult(timestamp);
+                        }
+                    });
+                    return;
+                }
+
+                finishWithResult(time.timestamp);
+            }
+        });
+    }
+
+    private void finishWithResult(long timestamp) {
+        Intent result = new Intent();
+        result.putExtra(ChooseFolder.EXTRA_ACCOUNT, mAccount.getUuid());
+        result.putExtra(EXTRA_SNOOZE_UNTIL, timestamp);
+        result.putExtra(ChooseFolder.EXTRA_MESSAGE, mMessageReference);
+        setResult(RESULT_OK, result);
+        finish();
+    }
+
+    private List<SnoozeTime> getTimes() {
+        List<SnoozeTime> times = new LinkedList<>();
+
+        long now = System.currentTimeMillis();
+        Calendar cal = new GregorianCalendar();
+        if (DEBUG) {
+            times.add(new SnoozeTime(now + TimeUnit.MINUTES.toMillis(1)));
+        }
+
+        times.add(new SnoozeTime(now + TimeUnit.HOURS.toMillis(1)));
+
+        // tomorrow
+        cal.add(Calendar.DAY_OF_YEAR, 1);
+        cal.set(Calendar.MINUTE, 0);
+        cal.set(Calendar.SECOND, 0);
+        cal.set(Calendar.HOUR_OF_DAY, 9);
+
+        times.add(new SnoozeTime(formatTimeLabel(cal, R.string.time_tomorrow_morning), cal.getTimeInMillis()));
+
+        cal.set(Calendar.HOUR_OF_DAY, 13);
+        times.add(new SnoozeTime(formatTimeLabel(cal, R.string.time_tomorrow_afternoon), cal.getTimeInMillis()));
+
+        cal.set(Calendar.HOUR_OF_DAY, 20);
+        times.add(new SnoozeTime(formatTimeLabel(cal, R.string.time_tomorrow_night), cal.getTimeInMillis()));
+
+        // next monday morning
```

Comment:
I did try `getFirstDayOfWeek()`, however it seemed to return "Sunday" for me instead of "Monday" (I'm in U.S.). But I feel that most people would want to wait until the start of the work week before they tackle emails. Do you know of a way to get a localised version of the first day of the *work* week?